atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/bucketingthings/ObjectPersisterTest.java

```diff
@@ -0,0 +1,82 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.atlasdb.sweep.asts.bucketingthings.ObjectPersister.LogSafety;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+import org.immutables.value.Value;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public final class ObjectPersisterTest {
+    @ParameterizedTest
+    @MethodSource("safety")
+    public void failedSerializationWithLogSafetyThrowsExceptionWithCorrectArgSafety(LogSafety safety, BiFunction<String, Object, Arg<?>> argFactory) {
+        ObjectPersister<NoJacksonAnnotations> persister = ObjectPersister.of(ObjectMappers.newServerSmileMapper(), NoJacksonAnnotations.class, safety);
+        NoJacksonAnnotations object = NoJacksonAnnotations.of();
+        assertThatLoggableExceptionThrownBy(() -> persister.trySerialize(object))
+                .hasLogMessage("Failed to serialize {} from {}")
+                .hasExactlyArgs(argFactory.apply("object", object), SafeArg.of("class", NoJacksonAnnotations.class.getName()));
+    }
+
+    @ParameterizedTest
+    @MethodSource("safety")
+    public void failedDeserializationWithLogSafetyThrowsExceptionWithCorrectArgSafety(LogSafety safety, BiFunction<String, Object, Arg<?>> argFactory) {
+        ObjectPersister<NoJacksonAnnotations> persister = ObjectPersister.of(ObjectMappers.newServerSmileMapper(), NoJacksonAnnotations.class, safety);
+        byte[] bytes = new byte[] {1, 2, 3};
+        assertThatLoggableExceptionThrownBy(() -> persister.tryDeserialize(bytes))
+                .hasLogMessage("Failed to deserialise {} into {}")
+                .hasExactlyArgs(argFactory.apply("bytes", bytes), SafeArg.of("class", NoJacksonAnnotations.class.getName()));
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = {0, 1, Long.MAX_VALUE, Long.MIN_VALUE}) // arbitrary
+    public void deserialisationIsInverseOfSerialisation(long value) {
+        ObjectPersister<Long> persister = ObjectPersister.of(ObjectMappers.newServerSmileMapper(), Long.class, LogSafety.SAFE);
+        assertThat(persister.tryDeserialize(persister.trySerialize(value))).isEqualTo(value);
+    }
+
+    private static Stream<Arguments> safety() {
+        // These need to be typed to use in Arguments
+        BiFunction<String, Object, Arg<?>> safeArgFactory = SafeArg::of;
+        BiFunction<String, Object, Arg<?>> unsafeArgFactory = UnsafeArg::of;
+        return Stream.of(
+                Arguments.of(LogSafety.SAFE, safeArgFactory),
+                Arguments.of(LogSafety.UNSAFE, unsafeArgFactory));
```

Comment:
optional: My preference here is probably to have an explicit parameter object, though this is fine.